### ISSUE TITLE
[v0.10.0.pre] Precompile Fails After Upgrade

### ISSUE THREAD

**Issue by jonmchan** [TrestleAdmin/trestle#467]
I didn't dive too deeply into this, just did a `bundle update` to the latest TrestleAdmin 0.10.0.pre. When I ran my CD build scripts, trestleadmin dies with the following error:

```
[7/7] RUN SECRET_KEY_BASE_DUMMY=1 CI=true bundle exec rails assets:precompile:
bin/rails aborted!
NameError: uninitialized constant Trestle::Resource::Controller (NameError)
 
        Trestle::Resource::Controller.send(:prepend, Trestle::Search::Controller)
                         ^^^^^^^^^^^^
/app/config/environment.rb:5:in `<main>'
Tasks: TOP => environment
(See full trace by running task with --trace)
```

It could be a me issue or it could be an issue with a downstream gem I'm using? It would seem that https://github.com/TrestleAdmin/trestle-search is breaking with the 0.10.0.pre upgrade.

**Comment by jonmchan** [comment]
Updated to trestle-search pre-release resolved above issue. Getting weird

```
SassC::SyntaxError: Internal Error: Not enough space (SassC::SyntaxError)
```

issue now. Investigating, will update on fixes.

**Comment by spohlenz** [comment]
I recall this one biting me a few times in the past. I believe the solution has been to:

1) Remove or rename any unnecessary `.scss` files under `app/assets/stylesheets/trestle/`. See the [0.10.0.pre release notes](https://github.com/TrestleAdmin/trestle/releases/tag/v0.10.0.pre) for info on how to do theming in 0.10.x.

2) Clear out your Sprockets cache by running `bundle exec rails tmp:cache:clear`.

**Comment by jonmchan** [comment]
Unfortunately that did not help.

I cleared the cache and removed everything from the app/assets/stylesheets folder just to test if it would work with nothing there, it still fails. I disabled all the add-on gems that are related to trestle like trestle-search, it still doesn't work.

In case it matters, this is how I am building my project:

```
RAILS_ENV=production SECRET_KEY_BASE_DUMMY=1 CI=true bundle exec rails assets:precompile
```

When I get rid of production:

```
SECRET_KEY_BASE_DUMMY=1 CI=true bundle exec rails assets:precompile
```

It works with no issue. Not sure what is the difference between production and non-production in this case.

**Comment by spohlenz** [comment]
Are you using Sass for your own stylesheets? If yes, then I have found [dartsass-sprockets](https://github.com/tablecheck/dartsass-sprockets) to be a decent drop-in alternative for `sassc-rails`.

If not, then you should be able to remove sassc-rails from your Gemfile, at least to see if it helps.

**Comment by jonmchan** [comment]
I had a few struggles with some outdated gems still depending on sassc-rails, but I forked the projects and changed their dependency to dartsass-sprockets or updated to newer version of the gem that removed the sassc dependency. I can confirm that upgrading to dartsass-sprockets does resolve this issue.

I can close this issue, but I would caution that v0.10.0.pre seems to have a problem with sassc-rails. (I don't understand how or why, as I am building the project from a clean github actions runner and I see that trestle 0.10.0.pre doesn't have any scss files -it doesn't completely make sense).

Unfortunately, there are still some not-so-well-maintained gems that may still have dependencies on sassc-rails. It might be good to make sure that trestle is able to continue playing nice with this legacy gem. I'll leave these findings here in case someone else runs into similar issues.

Thanks for your help @spohlenz !

**Comment by spohlenz** [comment]
I will definitely try to track down what in particular within the Trestle CSS is causing the Sass compilation error. I am able to reproduce it at least on a new Rails app after adding in the `sassc-rails` gem.

From a small amount of sleuthing, I've been able to determine that the error is happening in production mode but not development mode due to this line in sassc-rails: https://github.com/sass/sassc-rails/blob/master/lib/sassc/rails/railtie.rb#L74

I should also mention that another alternative to the `dartsass-sprockets` gem is to add the [sassc-embedded](https://github.com/sass-contrib/sassc-embedded-shim-ruby) gem to your Gemfile alongside `sassc-rails`:

```ruby
gem 'sassc-rails'
gem 'sassc-embedded'
```

Thank you for the issue report.

**Comment by spohlenz** [comment]
While the recommendations above are still the preferred option, I've added a fix in f10c78dff37337a931c5bd7ec37b8ddb7ef81f73 to prevent the errors when using `sassc-rails`.
